File: eCinema/eCinema.Model/Entities/Payment.cs
using eCinema.Models.Entities;
using System;

namespace CinemaApp.Domain.Entities
{
    public class Payment
    {
        public int Id { get; set; }
        public int BookingId { get; set; }
        public Booking Booking { get; set; }

        public long Amount { get; set; }
        public string Currency { get; set; } = "bam";

        public string StripePaymentIntentId { get; set; }
        public string StripeChargeId { get; set; }

        public PaymentStatus Status { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? SucceededAt { get; set; }
        public DateTime? FailedAt { get; set; }
        public DateTime? RefundedAt { get; set; }
    }

    public enum PaymentStatus
    {
        Pending = 0,
        Succeeded = 1,
        Failed = 2,
        Refunded = 3
    }

}

File: eCinema/eCinema.UI/ecinema_desktop/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';
part 'payment.g.dart';

enum PaymentStatus { pending, succeeded, failed, refunded, unknown }

PaymentStatus paymentStatusFromJson(dynamic value) {
  if (value == null) return PaymentStatus.unknown;

  if (value is int) {
    return PaymentStatus.values.asMap().containsKey(value)
        ? PaymentStatus.values[value]
        : PaymentStatus.unknown;
  }

  if (value is String) {
    switch (value.toLowerCase()) {
      case 'pending':
        return PaymentStatus.pending;
      case 'succeeded':
        return PaymentStatus.succeeded;
      case 'failed':
        return PaymentStatus.failed;
      case 'refunded':
        return PaymentStatus.refunded;
    }
  }
  return PaymentStatus.unknown;
}

String paymentStatusToJson(PaymentStatus s) => s.name;

@JsonSerializable(explicitToJson: true)
class Payment {
  final int id;
  final int bookingId;
  final num amount;
  final String currency;
  final String stripePaymentIntentId;
  final String? stripeChargeId;

  @JsonKey(fromJson: paymentStatusFromJson, toJson: paymentStatusToJson)
  final PaymentStatus status;

  final DateTime createdAt;
  final DateTime? succeededAt;
  final DateTime? failedAt;
  final DateTime? refundedAt;

  Payment({
    required this.id,
    required this.bookingId,
    required this.amount,
    required this.currency,
    required this.stripePaymentIntentId,
    this.stripeChargeId,
    required this.status,
    required this.createdAt,
    this.succeededAt,
    this.failedAt,
    this.refundedAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);
  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
